[Lum Wee Kiat] iP

weekiat-douze opened the pull request:
# DukePro
> "Your mind is for having ideas, not holding them." --David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's, 

* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/weekiat-douze/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :grimacing:

And it is **FREE**!

Features:

- [x] Managing Tasks
- [ ] Managing Deadlines (coming soon!)
- [ ] Reminders (coming soon!)

---
If you are a Java programmer, you can use it to practice Java too!
Here's the `main` method
```
    public static void main(String[] args) {
        new Duke("data/duke.txt").run();
    }

```

Commit:
Add Gradle support

weekiat-douze committed:
Completed Level-1: Greet, Echo, Exit

weekiat-douze committed:
Completed Level-2: Add, List

weekiat-douze committed:
Completed Level-3: Mark as Done

weekiat-douze committed:
Complete Level-4: ToDo, Event, Deadline

weekiat-douze committed:
Completed: A-TextUiTesting: Automated Text UI Testing

weekiat-douze committed:
Completed Level-5: Handle Errors, added DukeException.java

weekiat-douze committed:
Completed Level-6: Delete

weekiat-douze committed:
Cleaned up Main method by moving commands into their own methods

weekiat-douze committed:
Completed A-Enums: Use Enums

weekiat-douze committed:
Update command validation

weekiat-douze committed:
change switch-case in accordance with style guide

weekiat-douze committed:
Complete Level-7

dohaduong commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,170 @@
+        dukeSpeak("Bye. Hope to see you again soon!");
+
+    }
+    public static void todo(String[] input) throws DukeException {
+        if (input.length == 1 || input[1].isEmpty()) {
+            throw new DukeException(" ☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        Todo tempTodo = new Todo(input[1]);
+        taskList.add(tempTodo);
+        String message = "Got it. I've added this task:\n " + tempTodo.toString();
+        message += "\nNow you have " + taskList.size() + " tasks in the list.";
+        dukeSpeak(message);
+    }
```
these todo/deadline/event methods maybe you could use more OOP like Ui, Parser,... (according to the website). but it looks good anyways!

dohaduong commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task {
+    protected String from;
+    protected  String to;
```
Just a suggestion but maybe you could rename these to "startDate" "endDate" so it's clearer :)

dohaduong reviewed (commented):
Looks good!

weekiat-douze committed:
Complete Level-8: Dates and Times

weekiat-douze committed:
Merge branch 'branch-Level-7'

weekiat-douze committed:
Merge branch 'branch-Level-8'

nt-nic commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,11 @@
+public class Todo extends Task {
```
Good use of inheritance and subclasses to implement the different tasks

nt-nic commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,343 @@
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("____________________________________________________________\n" + logo);
```
Might be possible to use a constant variable for the horizontal lines

weekiat-douze committed:
Fix merge bugs: event save incorrect date and update text-ui-test

weekiat-douze committed:
Change variables and methods access to private

weekiat-douze committed:
Change date variable names in Deadline and Event (recommended by PR Reviews)

weekiat-douze committed:
Complete A-MoreOOP

weekiat-douze committed:
Complete A-Packages: Organize into Packages

weekiat-douze committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

weekiat-douze committed:
Complete A-JUnit

weekiat-douze committed:
Add file: META-INF/MANIFEST.MF

weekiat-douze committed:
Complete A-JavaDoc

weekiat-douze committed:
Complete A-CodingStandard

weekiat-douze committed:
Complete Level-9

weekiat-douze committed:
Merge branch 'branch-A-CodingStandard'

weekiat-douze committed:
Merge branch 'branch-Level-9'

weekiat-douze committed:
Merge branch 'branch-A-Javadoc'

# Conflicts:
#	src/main/java/duke/repository/Storage.java

weekiat-douze committed:
Complete A-CheckStyle

weekiat-douze committed:
Complete Level-10

weekiat-douze committed:
Revert "Complete Level-10"

This reverts commit dd0e99f89df3e2575af3b611cf027544360dc1a4.

weekiat-douze committed:
Revert "Revert "Complete Level-10""

This reverts commit 99b6f4c90018b68576aa30aa63e65e898428ea5e.

weekiat-douze committed:
Align DialogBox content

weekiat-douze committed:
Merge branch 'branch-Level-10'

weekiat-douze committed:
Cleanup new code according to coding standard and fix entry point in build.gradle

Some of the GUI related files (e.g. DialogBox.java, MainWindow.java) do not comply with coding standard.

Ran CheckStyle and fixed reported problems. Added JavaDocs to newly added Classes

weekiat-douze committed:
Complete A-Assertions

weekiat-douze committed:
Complete A-CodeQuality

Task adding methods in Duke class have common behaviours (e.g. construct message).
Methods in parser class checks for incorrect input using complicated expressions..
Duke class formats the string for saving before passing to Storage, causing different levels of abstractions.

Abstracted the message forming into a separate method. (generateAddTaskMessage and generateDeleteTaskMessage).
Extracted the if-clause conditions into a named boolean variable before using in the if-statement.
Abstracted save string formatting into a method in Storage.

Separating message forming in Duke class as it is reused across multiple methods, also ensure single level of abstraction (SLAP).
If-clause conditions are extracted into a single boolean instead of multiple as both are checking for insufficient input, also separating will cause error as they are dependent.
Abstracted into Storage to ensure SLAP in both Duke methods and saveEntry method in Storage.

weekiat-douze committed:
Merge pull request #2 from weekiat-douze/branch-A-Assertions

Complete A-Assertions

weekiat-douze committed:
Merge branch 'master' into branch-A-CodeQuality

weekiat-douze committed:
Merge pull request #3 from weekiat-douze/branch-A-CodeQuality

Complete A-CodeQuality

weekiat-douze committed:
Complete C-MassOps

Currently Mark, Unmark, Delete action only support 1 tasks at a time.

Adding MassOps allows user to do above actions for multiple tasks in a single command.

Changing relevant methods in Storage, Parser and TaskList to accept/return ArrayList of tasks.

ArrayList size is variable so user can have variable number of tasks to act on.
Mark, Unmark, Delete on single task is still supported

weekiat-douze committed:
Merge branch 'branch-C-MassOps'

weekiat-douze committed:
Change wording for Mass-Ops

weekiat-douze committed:
Complete A-Personality

weekiat-douze committed:
Fix wrong deletion in TaskList

weekiat-douze committed:
Add title interface, update duke and user profile picture

weekiat-douze committed:
Complete A-UserGuide

weekiat-douze committed:
Fix user guide divider

weekiat-douze committed:
Fix spelling error in User Guide
